src/main/java/uk/gov/companieshouse/account/validator/controller/AccountValidationController.java

```diff
@@ -95,6 +111,40 @@ ResponseEntity<?> getStatus(@PathVariable final String fileId) {
         return ValidationResponse.success(requestStatus.get());
     }
 
+    /**
+     * Handles the request to transform a iXBRL file stored in S3 to PDF
+     *
+     * @param fileId of remote file
+     * @return S3 file as PDF
+     */
+    @GetMapping(path = "/render/{fileId}")
+    public ResponseEntity<?> render(@PathVariable String fileId) {
+        var file = fileTransferStrategy.get(fileId);
+        if (file.isEmpty()) {
+            return ValidationResponse.fileNotFound();
+        }
+
+        String iXbrlToPdfUri = getIxbrlToPDFEnvVal();
+        if (StringUtils.isBlank(iXbrlToPdfUri)) {
+            throw new MissingEnvironmentVariableException(String.format("Missing '%s' environment variable", IXBRL_TO_PDF_URI_KEY));
+        }
+
+        ByteArrayResource contentsAsResource = new ByteArrayResource(file.get().getData()) {
+            @Override
+            public String getFilename() {
+                return "anything";
```

Comment:
Is this "anything" going to be changed to a correct string?

Reply:
no, doesn't make any difference, required when posting byte[] as per https://medium.com/@voziv/posting-a-byte-array-instead-of-a-file-using-spring-s-resttemplate-56268b45140b - stumped me for quite a while